src/main/java/Rei.java

```diff
@@ -0,0 +1,168 @@
+
+        System.out.println("Annyeong! I'm\n" + logo);
+        System.out.println("What can I do for you?");
+        System.out.println("-----------YOU------------");
+
+        List<Task> listOfTasks = new ArrayList<>();
+        Scanner scanner = new Scanner(System.in);
+
+        while (!scanner.hasNext("annyeong") && !scanner.hasNext("Annyeong")) {
+            if (scanner.hasNext("list")) {
+                System.out.println("-----------REI♥-----------");
+                System.out.println("Here are the tasks in your list: ");
+                for (int i = 0; i < listOfTasks.size(); i++) {
+                    System.out.println((i + 1) + ". " + listOfTasks.get(i));
+                }
+                scanner.nextLine();
+            } else if (scanner.hasNext("mark")) {
+                System.out.println("-----------REI♥-----------");
+                scanner.next();
+
+                // Read the rest of the line after "mark"
+                String prompt = scanner.nextLine().trim();
+
+                // Check if the rest of the line is an integer
+                if (prompt.isEmpty() || !prompt.matches("\\d+")) {
+                    System.out.println("State the task number.");
+                    System.out.println("-----------YOU------------");
+                    continue;
+                }
+
+                int id = Integer.parseInt(prompt);
+                if (id <= listOfTasks.size() && id > 0) {
+                    listOfTasks.get(id - 1).markAsDone();
+                    System.out.println("Okay! I've marked this task as done: ");
+                    System.out.println(listOfTasks.get(id - 1));
+                } else {
+                    System.out.println("No task found. Please retry!");
+                }
+
+            } else if (scanner.hasNext("unmark")) {
+                System.out.println("-----------REI♥-----------");
+                scanner.next();
+
+                // Read the rest of the line after "unmark"
+                String prompt = scanner.nextLine().trim();
+
+                // Check if the rest of the line is an integer
+                if (prompt.isEmpty() || !prompt.matches("\\d+")) {
+                    System.out.println("State the task number.");
+                    System.out.println("-----------YOU------------");
+                    continue;
+                }
+
+                int id = Integer.parseInt(prompt);
+                if (id <= listOfTasks.size() && id > 0) {
+                    listOfTasks.get(id - 1).markAsUndone();
+                    System.out.println("Okay! I've marked this task as not done yet: ");
+                    System.out.println(listOfTasks.get(id - 1));
+                } else {
+                    System.out.println("No task found. Please retry!");
+                }
+            } else if (scanner.hasNext("todo")
+                        || scanner.hasNext("deadline")
+                        || scanner.hasNext("event")) {
```

Comment:
I like this grouping of commands: it makes more sense especially when you go deeper down the OOP rabbithole and have many classes representing different tasks. This inspires me to incorporate this into my code.

Reply:
Overall most method and variable names (eg. `isAllWhitespace`) are quite clear and descriptive, nice!

Reply:
Just a small nitpick but could consider changing `id` to something a bit clearer like `listIndex`?